### DISCUSSION TITLE
Add publication

### DISCUSSION THREAD

**Discussion by SomeUserName1** [jaxleyverse/jaxley#407] [Q&A]
Hi Michael,

Is there a paper or a preprint or something similar on Jaxley?

Can you shortly elaborate on the neuron model and the difference between Jaxley and NEURON?

Regarding NEURON, the book says:
>  The axon and dendrites are simple cylinders, with uniform diameters and membrane properties along their lengths. The dendrites are passive, while the soma and axon have Hodgkin-Huxley sodium, potassium, and leak currents, and are capable of generating action potentials (Hodgkin and Huxley 1952). 

However, that was written in 2004. 
From what I understand, NMODL allows for extensions, however just for ion channels.

Jaxley also implements an extensible HH-model, correct?
Is Jaxley in theory capable of supporting GPCRs in some way?
What about dendritic non-linearities, e.g. synaptic clustering, shunting, Ca-spikes and the like?

Thanks in advance
Fabian

**Comment by michaeldeistler** [comment]
Dear Fabian,

thanks for reaching out!

We are currently preparing the preprint for Jaxley and will submit it this week (that’s why we just made the code public). I will let you know once it is out!

Yes, (like NEURON), Jaxley also allows to implement any kind of custom ion channels, see [here](https://jaxleyverse.github.io/jaxley/tutorial/05_channel_and_synapse_models/).

And yes, (like NEURON), Jaxley allows to have channels in any kinds of neurite (e.g. only in the soma, or in the dendrite). See the `Insert mechanisms` subsection [here](https://jaxleyverse.github.io/jaxley/tutorial/01_morph_neurons/).

Because Jaxley allows to insert any kind of channel in any kind of neurite (e.g., dendrite), it can also model dendritic nonlinearities, shunting inhibition, calcium spikes, etc. You can find a collection of currently implemented channels [here](https://github.com/jaxleyverse/jaxley-mech/tree/main/jaxley_mech/channels).

What are GPCRs?

Hope this helps!

Michael

**Comment by michaeldeistler** [comment]
The big differences between Jaxley and NEURON are that:
1) Jaxley runs on CPU and GPU without any changes to the code
2) Jaxley allows to compute the gradient with respect to parameters

**Comment by huangziwei** [comment]
> GPCRs

G protein-coupled receptors? In Jaxley, we use the concept of `Synapse` instead of `Receptor`, which deviates slightly from the convention used in NEURON, but usage-wise it's actually very similar.

https://jaxleyverse.github.io/jaxley/tutorial/05_channel_and_synapse_models/#your-own-synapse

https://github.com/jaxleyverse/jaxley-mech/tree/main/jaxley_mech/synapses

**Comment by michaeldeistler** [comment]
The preprint for `Jaxley` is on biorxiv now, see [here](https://www.biorxiv.org/content/10.1101/2024.08.21.608979).
